mmdeploy/codebase/mmdet/deploy/object_detection.py

```diff
@@ -114,7 +114,23 @@ def create_input(self,
         from mmcv.parallel import collate, scatter
         if not isinstance(imgs, (list, tuple)):
             imgs = [imgs]
+        dynamic_flag = is_dynamic_shape(self.deploy_cfg)
         cfg = process_model_config(self.model_cfg, imgs, input_shape)
+        # Drop pad_to_square when static shape. Because static shape should
+        # ensure the shape before input image.
+        if not dynamic_flag:
+            transform = cfg.data.test.pipeline[1]
+            if 'transforms' in transform:
+                transform_list = transform['transforms']
+                ind = []
+                pos = 0
+                for step in transform_list:
```

Comment:
for i, step in enumerate(transform_list):
    if step['type'] == 'Pad' and 'pad_to_square' in step \
                       and step['pad_to_square']:
        transform_list.pop(i)
        break